inspirehep/modules/migrator/tasks.py

```diff
@@ -237,58 +241,47 @@ def get_records_to_update_generator(citation_lookup):
                     'doc': {'citation_count': citation_count}
                 }
 
-    click.echo("Extracting all citations...")
-
-    # lookup dictionary where key: recid of the record
-    # and value: number of records that cite that record
+    index, doc_type = schema_to_index('records/hep.json')
     citations_lookup = Counter()
+
+    click.echo('Extracting all citations...')
     with click.progressbar(es_scan(
             current_search_client,
             query={
-                "_source": "references.recid",
-                "filter": {
-                    "exists": {
-                        "field": "references.recid"
+                '_source': 'references.recid',
+                'filter': {
+                    'exists': {
+                        'field': 'references.recid'
                     }
                 },
-                "size": LARGE_CHUNK_SIZE
+                'size': LARGE_CHUNK_SIZE
             },
             scroll=u'2m',
             index=index,
             doc_type=doc_type)) as records:
         for record in records:
-            # update lookup dictionary based on references of the record
-            if 'references' in record['_source']:
-                unique_refs_ids = set()
-                references = record['_source']['references']
-                for reference in references:
-                    recid = reference.get('recid')
-                    if recid:
-                        if isinstance(recid, list):
-                            # Sometimes there is more than one recid in the
-                            # reference.
-                            recid = recid.pop()
-                        unique_refs_ids.add(recid)
+            unique_refs_ids = dedupe_list(list(chain.from_iterable(map(
+                force_force_list, get_value(record, '_source.references.recid')))))
```

Comment:
Mmh I am not really understanding how you have refactored, but I trust is working :+1: